pkg/watermark/fetch/edge_fetcher.go

```diff
@@ -205,10 +210,7 @@ func (e *edgeFetcher) GetHeadWMB(fromPartitionIdx int32) wmb.WMB {
 	defer e.Unlock()
 	// update the last processed watermark for the partition
 	e.lastProcessedWm[fromPartitionIdx] = headWMB.Watermark
-	// we only consider idle watermark if it is smaller than or equal to min of all the last processed watermarks.
-	if headWMB.Watermark > e.getMinFromLastProcessed(headWMB.Watermark) {
-		return wmb.WMB{}
-	}
```

Comment:
Add back the check

Reply:
Per our discussion I have temporarily added the check back and removed it from EdgeFetcherSet until the interfaces for the two structs can be split